serviceLibrary/src/main/AndroidManifest.xml

```diff
@@ -3,6 +3,7 @@
 
     <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE" />
     <uses-permission android:name="android.permission.INTERNET" />
+    <uses-permission android:name="android.permission.SCHEDULE_EXACT_ALARM" />
```

Comment:
Do you see an error without it ?

Reply:
![image](https://user-images.githubusercontent.com/2796439/136026541-aa98900d-99db-44fc-9e8d-722895f0bb7e.png)


	java.lang.SecurityException: Caller info.mqtt.java.example needs to hold android.permission.SCHEDULE_EXACT_ALARM to set exact alarms.
	        at android.os.Parcel.createExceptionOrNull(Parcel.java:2425)
	        at android.os.Parcel.createException(Parcel.java:2409)
	        at android.os.Parcel.readException(Parcel.java:2392)
	        at android.os.Parcel.readException(Parcel.java:2334)
	        at android.app.IAlarmManager$Stub$Proxy.set(IAlarmManager.java:359)
	        at android.app.AlarmManager.setImpl(AlarmManager.java:947)
	        at android.app.AlarmManager.setImpl(AlarmManager.java:907)
	        at android.app.AlarmManager.setExactAndAllowWhileIdle(AlarmManager.java:1175)
	        at info.mqtt.android.service.ping.AlarmPingSender.schedule(AlarmPingSender.kt:89)
	        at info.mqtt.android.service.ping.AlarmPingSender.start(AlarmPingSender.kt:60)
	        at org.eclipse.paho.client.mqttv3.internal.ClientState.connected(ClientState.java:1214)
	        at org.eclipse.paho.client.mqttv3.internal.ClientState.notifyReceivedAck(ClientState.java:1050)
	        at org.eclipse.paho.client.mqttv3.internal.CommsReceiver.run(CommsReceiver.java:151)
	        at java.lang.Thread.run(Thread.java:920)
	     Caused by: android.os.RemoteException: Remote stack trace:
	        at com.android.server.alarm.AlarmManagerService$5.set(AlarmManagerService.java:2534)
	        at android.app.IAlarmManager$Stub.onTransact(IAlarmManager.java:194)
	        at android.os.Binder.execTransactInternal(Binder.java:1179)
	        at android.os.Binder.execTransact(Binder.java:1143)
	E/(LoggingTools.kt:12) $globalErrorCatcher.uncaughtException(): android.os.RemoteException: Remote stack trace:
	        at com.android.server.alarm.AlarmManagerService$5.set(AlarmManagerService.java:2534)
	        at android.app.IAlarmManager$Stub.onTransact(IAlarmManager.java:194)
	        at android.os.Binder.execTransactInternal(Binder.java:1179)
	        at android.os.Binder.execTransact(Binder.java:1143)

Reply:
Thank you.

Btw, this is a weak point of this lib, I started to switch to `WorkMannager` https://github.com/hannesa2/paho.mqtt.android/pull/159 

But It's untested

Reply:
Ok we're looking at this